File: src/main/java/ch/obermuhlner/astro/image/ImageReader.java
package ch.obermuhlner.astro.image;

import mil.nga.tiff.TiffReader;

import javax.imageio.ImageIO;
import java.io.File;
import java.io.IOException;

public class ImageReader {

  public static DoubleImage read(File file) throws IOException {
    return read(file, ImageQuality.High);
  }

  public static DoubleImage read(File file, ImageQuality quality) throws IOException {
    if (quality == ImageQuality.High) {
      try {
        return new TiffDoubleImage(TiffReader.readTiff(file), true);
      } catch (Exception ex) {
        // ignore
      }
    }

    return new AwtBufferedDoubleImage(ImageIO.read(file));
  }
}

File: src/main/java/ch/obermuhlner/astro/image/ImageReader.kt
package ch.obermuhlner.astro.image

import mil.nga.tiff.TiffReader
import java.io.File
import java.io.IOException
import javax.imageio.ImageIO

object ImageReader {
    @JvmOverloads
    @Throws(IOException::class)
    fun read(file: File, quality: ImageQuality = ImageQuality.High): DoubleImage {
        if (quality == ImageQuality.High) {
            try {
                return TiffDoubleImage(TiffReader.readTiff(file), true)
            } catch (ex: Exception) {
                // ignore
            }
        }
        return AwtBufferedDoubleImage(ImageIO.read(file))
    }
}